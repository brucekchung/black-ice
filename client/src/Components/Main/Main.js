import React from 'react'
import { Switch, Route } from 'react-router-dom'

import Login from '../Login/Login'
import DataEntry from '../DataEntry/DataEntry'
import Reports from '../Reports/Reports'
import Landing from '../Landing/Landing'
import ViewAll from '../ViewAll/ViewAll'

const Main = () => (
  <Switch>
    <Route exact path='/' render={ () => <Login /> } /> 
    <Route exact path='/DataEntry' render={ () => <DataEntry /> } /> 
    <Route exact path='/Reports' render={ () => <Reports /> } /> 
    <Route exact path='/Landing' render={ () => <Landing /> } /> 
    <Route exact path='/ViewAll' render={ () => <ViewAll /> } /> 
  </Switch>
)

export default Main
