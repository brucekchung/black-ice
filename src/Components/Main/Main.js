import React from 'react'
import { Switch, Route } from 'react-router-dom'

import Login from '../Login/Login'
const Main = () => (
  <Switch>
    <Route exact path='/' render={ () => <Login /> } /> 
  </Switch>
)

export default Main
