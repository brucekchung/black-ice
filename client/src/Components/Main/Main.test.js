import React from 'react'
import Main from './Main'
import { shallow } from 'enzyme'

describe('Main', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Main />)
  })

  it('should match snapshot upon load', () => {
    expect(wrapper).toMatchSnapshot()
  })

})