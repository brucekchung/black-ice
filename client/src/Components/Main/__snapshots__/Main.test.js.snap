// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot upon load 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/DataEntry"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/Reports"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/Landing"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/ViewAll"
          render={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/DataEntry",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/Reports",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/Landing",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/ViewAll",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/DataEntry"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/Reports"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/Landing"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/ViewAll"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/DataEntry",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/Reports",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/Landing",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/ViewAll",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
